read-fonts/src/tables/postscript/scale.rs

```diff
@@ -0,0 +1,541 @@
+            // Missing FDSelect assumes either a single Font DICT at index 0,
+            // or a Private DICT entry in the Top DICT.
+            .unwrap_or(0) as u32
+    }
+
+    /// Creates a new subfont for the given index, size, normalized
+    /// variation coordinates and hinting state.
+    ///
+    /// The index of a subfont for a particular glyph can be retrieved with
+    /// the [`subfont_index`](Self::subfont_index) method.
+    pub fn subfont(
+        &self,
+        index: u32,
+        size: f32,
+        coords: &[F2Dot14],
+        with_hinting: bool,
+    ) -> Result<ScalerSubfont, Error> {
+        let private_dict_range = self.private_dict_range(index)?;
+        let private_dict_data = self.offset_data().read_array(private_dict_range.clone())?;
+        let mut hint_params = HintParams::default();
+        let mut subrs_offset = None;
+        let mut store_index = 0;
+        let blend_state = self
+            .top_dict
+            .var_store
+            .clone()
+            .map(|store| BlendState::new(store, coords, store_index))
+            .transpose()?;
+        for entry in dict::entries(private_dict_data, blend_state) {
+            use dict::Entry::*;
+            match entry? {
+                BlueValues(values) => hint_params.blues = values,
+                FamilyBlues(values) => hint_params.family_blues = values,
+                OtherBlues(values) => hint_params.other_blues = values,
+                FamilyOtherBlues(values) => hint_params.family_blues = values,
+                BlueScale(value) => hint_params.blue_scale = value,
+                BlueShift(value) => hint_params.blue_shift = value,
+                BlueFuzz(value) => hint_params.blue_fuzz = value,
+                LanguageGroup(group) => hint_params.language_group = group,
+                // Subrs offset is relative to the private DICT
+                SubrsOffset(offset) => subrs_offset = Some(private_dict_range.start + offset),
+                VariationStoreIndex(index) => store_index = index,
+                _ => {}
+            }
+        }
+        // TODO: convert hint params to zones if hinting is requested
+        let _ = with_hinting;
+        Ok(ScalerSubfont {
+            is_cff2: self.is_cff2(),
+            index,
+            size,
+            subrs_offset,
+            hint_params,
+            store_index,
+        })
+    }
+
+    /// Loads and scales an outline for the given subfont instance, glyph
+    /// identifier and normalized variation coordinates.
+    ///
+    /// Before calling this method, use [`subfont_index`](Self::subfont_index)
+    /// to retrieve the subfont index for the desired glyph and then
+    /// [`subfont`](Self::subfont) to create an instance of the subfont for a
+    /// particular size and location in variation space.
+    /// Creating subfont instances is not free, so this process is exposed in
+    /// discrete steps to allow for caching.
```

Comment:
This seems like it's exposing exactly the sort of detail Context claims to abstract away from me? - not necessarily a problem but perhaps we should explain why.